src/crawlers/browser_crawler.js

```diff
@@ -360,8 +360,43 @@ export default class BrowserCrawler extends BasicCrawler {
      * @internal
      */
     async _handleRequestFunction(crawlingContext) {
-        const { id } = crawlingContext;
-        const page = await this.browserPool.newPage({ id });
+        const newPageOptions = {
+            id: crawlingContext.id,
+        };
+
+        if (this.proxyConfiguration && this.launchContext.useIncognitoPages) {
```

Comment:
I also don't like the fact that `this.launchContext` isn't defined in `BrowserCrawler`

Reply:
Probably an omission. It would make sense to have it in `BrowserCrawler`.